lib/active_set/sorting/operation.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+require_relative '../attribute_instruction'
+
+class ActiveSet
+  module Sorting
+    class Operation
+      def initialize(set, instructions_hash)
+        @set = set
+        @instructions_hash = instructions_hash
+      end
+
+      def execute
+        attribute_instructions = @instructions_hash
+                                   .flatten_keys
+                                   .map { |k, v| AttributeInstruction.new(k, v) }
+
+        activerecord_strategy = ActiveRecordStrategy.new(@set, attribute_instructions)
+        if activerecord_strategy.executable_instructions == attribute_instructions
+          activerecord_sorted_set = activerecord_strategy.execute
+        end
+
+        return activerecord_sorted_set if attribute_instructions.all?(&:processed?)
+
+        EnumerableStrategy.new(@set, attribute_instructions).execute
+      end
+
+      def operation_instructions
+        @instructions_hash.symbolize_keys
+      end
+    end
+
+    class EnumerableStrategy
+      def initialize(set, attribute_instructions)
+        @set = set
+        @attribute_instructions = attribute_instructions
+      end
+
+      def execute
+        @set.sort_by do |item|
```

Comment:
`sort_by` uses a _Schwartzian Transform_ for sorting, which is more efficient for large collections (see http://brandon.dimcheff.com/2009/11/18/rubys-sort-vs-sort-by/). However, in order to use it, we need to normalize the values per item that we sort by, and thus we need to find sortable, numeric values for each attribute.